frontend/src/map.jsx

```diff
@@ -97,6 +160,13 @@ export default function MapComponent(props) {
     map.on(['click'], handleClick);
   }, []);
 
+  useEffect(() => {
+    if (map) {
+      const patternSamplerLayer = map.getLayers().getArray()[3];
```

Comment:
I wonder if this is a reason to move `patternSamplerLayer` to it's own module to import. That would keep it in scope to use in this function without having to find it in the list of layers. And then maybe we don't need `map` in state at all?

I guess the `Polygon` would still need to be created in this component, where it has access to the map View. But I assume we can do something like `patternSamplerLayer.addFeature()` to add it after the layer was constructed.